org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/crawler/GithubCrawler.java

```diff
@@ -0,0 +1,232 @@
+                        successful = false;
+                        if (isSearch) {
+                            rateLimitResetSearch = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+                            connection.disconnect();
+                            sleepSearchRateLimit();
+                        } else {
+                            rateLimitReset = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+                            connection.disconnect();
+                            sleepRateLimit();
+                        }
+                    } else {
+                        fails++;
+                        successful = false;
+                        System.err.println("An error occurred in a http get request: " + connection.getResponseCode() + " - " + connection.getResponseMessage() + ". Failed " + fails + ". time.");
+                        sleepOneMinute();
+                    }
+                } else {
+                    fails++;
+                    successful = false;
+                    System.err.println("An error occurred in a http get request: " + connection.getResponseCode() + " - " + connection.getResponseMessage() + ". Failed " + fails + ". time.");
+                    sleepOneMinute();
+                }
+            } else {
+                successful = true;
+            }
+        } while (!successful);
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+
+        if (isInRateLimit && isSearch) {
+            rateLimitRemainingSearch = Integer.parseInt(connection.getHeaderField("X-RateLimit-Remaining"));
+            rateLimitResetSearch = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+            System.out.println("Search rate limit: " + rateLimitRemainingSearch);
+        } else if (isInRateLimit) {
+            rateLimitRemaining = Integer.parseInt(connection.getHeaderField("X-RateLimit-Remaining"));
+            rateLimitReset = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+            System.out.println("Rate limit: " + rateLimitRemaining);
+        }
+        String line;
+        while ((line = reader.readLine()) != null) {
+            responseBuilder.append(line).append("\n");
+        }
+        reader.close();
+
+        return responseBuilder.toString();
+    }
+
+    private void sleepSearchRateLimit() {
+        long sleepTime = rateLimitResetSearch.toEpochMilli() - Instant.now().toEpochMilli() + 5000;
+        if (sleepTime > 0) {
+            System.out.println("Rate Limit: GitHub search api rate limit exceeded! Thread sleeps " + sleepTime + " milliseconds.");
+            try {
+                Thread.sleep(sleepTime);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
`Logger.error("description", e)`

I won't mention this here again. Please check all files by yourself and adjust them accordingly.